File: docs/sphinx/targets/cpp/tii.cpp
// 
// Available projects and devices are show in the dashboard:
// https://q-cloud.tii.ae/projects/
//
// The authentication key must be stored (or exported) as enivronment variable:
// ```
// export TII_API_TOKEN="you-tii-token"
// ```
//
// Compile and run with:
// ```
// nvq++ --target tii --tii-device tii-sim \
// --tii-project <tii-project-name> tii.cpp
// ./a.out
// ```

#include <cudaq.h>
#include <fstream>

struct bell_state {
  auto operator()() __qpu__ {
    cudaq::qvector q(2);
    h(q[0]);
    x<cudaq::ctrl>(q[0], q[1]);
    mz(q);
  }
};

int main() {
  // Submit to `tii` synchronously.
  auto result_counts = cudaq::sample(bell_state{});
  result_counts.dump();
}

File: docs/sphinx/targets/python/tii.py
import cudaq
import os

# Set the target at the beginning of the program.
cudaq.set_target("tii",
                 device="tii-sim",
                 project=os.environ.get("TII_PROJECT", None))


# Create the kernel.
@cudaq.kernel
def kernel():
    qvector = cudaq.qvector(2)
    h(qvector[0])
    x.ctrl(qvector[0], qvector[1])
    mz(qvector)


# Execute on synchronously on the TII cloud and print out the results.
counts = cudaq.sample(kernel)
print(counts)
